src/nvim/path.c

```diff
@@ -360,7 +360,7 @@ void add_pathsep(char *p)
   FUNC_ATTR_NONNULL_ALL
 {
   if (*p != NUL && !after_pathsep(p, p + strlen(p)))
-    strcat(p, PATHSEPSTR);
+    STRLCAT(p, PATHSEPSTR, MAXPATHL);
```

Comment:
Can you add a test which created directory `x…x` (255 characters) and uses various functions which use `add_pathsep` on this path? `cd x…x | echo fnamemodify('', ':p')` should work.